js/pathfinding.js

```diff
@@ -55,27 +53,51 @@ export function findPath(from, to, token, previousWaypoints) {
 	}
 }
 
-function getNode(pos, token, initialize=true) {
-	if (!cachedNodes)
-		cachedNodes = new Array(gridHeight);
-	if (!cachedNodes[pos.y])
-		cachedNodes[pos.y] = new Array(gridWidth);
-	if (!cachedNodes[pos.y][pos.x]) {
-		cachedNodes[pos.y][pos.x] = pos;
+/**
+ * Build a cache ID based on the current token's data and then retrieve the cache to use from that
+ */
+function getCachedNodes(token) {
+	const cacheData = {};
+
+	// Different-sized tokens snap to different points on the grid,
+	// so they might follow a different path to other tokens
+	cacheData.tokenWidth = token.width;
+	cacheData.tokenHeight = token.height;
```

Comment:
On hex grids (with the `hex-size-support` module enabled), a tokens size works a little differently. The token has a numeric size and a shape inversion flag. You either need to encode those or you could just take the array returned by `getTokenShape` as indicator.

Reply:
Updated so we only consider token width and height for square grid, and use token size on a hex grid if `hex-size-support` is enabled.